src/Command/BaseCommand.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace SilverStripe\DeprecationChecker\Command;
+
+use RuntimeException;
+use stdClass;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Helper;
+use Symfony\Component\Console\Helper\ProcessHelper;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Console\Terminal;
+use Symfony\Component\Process\Process;
+
+/**
+ * Base class for commands - provides some helper methods and a nicer default output.
+ */
+abstract class BaseCommand extends Command
+{
+    protected InputInterface $input;
+
+    protected SymfonyStyle $output;
+
+    private ?ProgressBar $progressBar = null;
+
+    private bool $progressBarDisplayed = false;
+
+    /**
+     * Callable to pass into Process::run() to advance a progress bar when a long process
+     * has some output
+     */
+    public function handleProcessOutput($type, $data): void
```

Comment:
Strongly type params

Reply:
I have added types that I hope are correct, though since this is used as a callback in code we don't control, I can't be held responsible if some other type is passed in :p